app/actions/websocket/index.ts

```diff
@@ -64,90 +68,119 @@ export async function handleClose(serverUrl: string, lastDisconnect: number) {
-
-    // TODO consider fetch only and batch all the results.
-    await fetchMe(serverUrl);
-    await fetchMyPreferences(serverUrl);
-    const {config} = await fetchConfigAndLicense(serverUrl);
-    const {memberships: teamMemberships, error: teamMembershipsError} = await fetchMyTeams(serverUrl);
-    const {currentChannelId, currentUserId, currentTeamId, license} = system;
-    const currentTeamMembership = teamMemberships?.find((tm) => tm.team_id === currentTeamId && tm.delete_at === 0);
-
-    let channelMemberships: ChannelMembership[] | undefined;
-    if (currentTeamMembership) {
-        const {memberships, channels, error} = await fetchMyChannelsForTeam(serverUrl, currentTeamMembership.team_id, true, lastDisconnectedAt);
-        if (error) {
-            DeviceEventEmitter.emit(Events.TEAM_LOAD_ERROR, serverUrl, error);
-            return;
-        }
-        const currentUser = await queryCurrentUser(database.database);
-        const preferences = currentUser ? (await currentUser.preferences.fetch()) : [];
-        const teammateDisplayNameSetting = getTeammateNameDisplaySetting(preferences || [], system.config, license);
-        const directChannels = channels?.filter((c) => c.type === General.DM_CHANNEL || c.type === General.GM_CHANNEL);
+    const {database} = operator;
+    const tabletDevice = await isTablet();
+    const system = await queryCommonSystemValues(database);
+    const lastDisconnectedAt = await queryWebSocketLastDisconnected(database);
+
+    resetWebSocketLastDisconnected(operator);
+    let {config, license} = await fetchConfigAndLicense(serverUrl);
+    if (!config) {
+        config = system.config;
+    }
+
+    if (!license) {
+        license = system.license;
+    }
+
+    const fetchedData = await fetchAppEntryData(serverUrl, lastDisconnectedAt, system.currentTeamId);
+    const fetchedError = (fetchedData as AppEntryError).error;
+
+    if (fetchedError) {
+        return;
+    }
+
+    const {initialTeamId, teamData, chData, prefData, meData, removeTeamIds, removeChannelIds} = fetchedData as AppEntryData;
+    const rolesData = await fetchRoles(serverUrl, teamData.memberships, chData?.memberships, meData.user, true);
+
+    if (chData?.channels?.length) {
+        const teammateDisplayNameSetting = getTeammateNameDisplaySetting(prefData.preferences || [], config, license);
+        const directChannels = chData.channels.filter((c) => c.type === General.DM_CHANNEL || c.type === General.GM_CHANNEL);
+        chData.channels = chData.channels.filter((c) => c.type !== General.DM_CHANNEL && c.type !== General.GM_CHANNEL) || [];
+
         if (directChannels?.length) {
-            await fetchMissingSidebarInfo(serverUrl, directChannels, currentUser?.locale, teammateDisplayNameSetting, currentUserId);
+            await fetchMissingSidebarInfo(serverUrl, directChannels, meData.user?.locale, teammateDisplayNameSetting, system.currentUserId, true);
+            chData.channels.push(...directChannels);
```

Comment:
This seems not to do anything unless we assign the value of the function in the previous line to directChannels.